go/vt/vtgate/safe_session.go

```diff
@@ -74,18 +89,36 @@ func NewAutocommitSession(sessn *vtgatepb.Session) *SafeSession {
 	newSession := proto.Clone(sessn).(*vtgatepb.Session)
 	newSession.InTransaction = false
 	newSession.ShardSessions = nil
+	newSession.PreSessions = nil
+	newSession.PostSessions = nil
 	newSession.Autocommit = true
+	newSession.Warnings = nil
```

Comment:
Why are the warnings cleared here?

Reply:
This is used as a private session within the context of an outer session, specifically for vindex autocommit. The warnings of the outer one are untouched. After this execution finishes, we could consider appending any new warnings generated back to the outer session. But that may or may not be right. We should look at some real use cases before making this decision.

Maybe we should rename this to `NewInnerSession`?